cpp/src/parquet/decoder.cc

```diff
@@ -178,6 +178,9 @@ struct ArrowBinaryHelper<ByteArrayType, ArrowBinaryType> {
 
   Status AppendValue(const uint8_t* data, int32_t length,
                      std::optional<int64_t> estimated_remaining_data_length = {}) {
+    if (ARROW_PREDICT_FALSE(data == nullptr)) {
```

Comment:
Perhaps we need find out in which case this is nullptr

Reply:
Have you investigated this @mapleFU ?